driver/normalizer/normalizer.go

```diff
@@ -1019,13 +1187,26 @@ func (op opMergeGroups) Kinds() nodes.Kind {
 	return nodes.KindObject
 }
 
-// Check tests if the current node is uast:Group and if it contains a uast:FunctionGroup
-// node, it will remove the current node and merge other children into the FunctionGroup.
+// Check tests if a current node is uast:Group and uast:FuncGroup and contains group of
+// another kind. It will remove the second group and merge children into current one.
+// uast:FuncGroup is preferred.
 func (op opMergeGroups) Check(st *State, n nodes.Node) (bool, error) {
 	group, ok := n.(nodes.Object)
-	if !ok || uast.TypeOf(group) != typeGroup {
+	if !ok {
 		return false, nil
 	}
+	switch uast.TypeOf(group) {
+	case typeGroup:
+		return op.checkGroup(st, group)
+	case typeFuncGroup:
+		return op.checkFuncGroup(st, group)
+	}
+	return false, nil
+}
+
+// checkGroup tests if the current node is uast:Group and if it contains a uast:FunctionGroup
```

Comment:
Thanks for adding the comment. I suggest going even a bit further to highlight the mutation: I found it surprising that a function that reports on the state of the value also changes the _form_ of the value. Maybe something like

```
// checkGroup reports whether the current node is a uast:Group. If so, and if it contains a uast:FunctionGroup 
// node, it will also replace the current node with the uast:FunctionGroup, and fold the other children of the
// original node into the FunctionGroup.
```

or words to that effect. (Separately, I think we ought to rethink this design a little, but that is a larger question and out-of-scope for this PR).